src/sentry/api/endpoints/issues/related_issues.py

```diff
@@ -33,6 +40,9 @@ def get(self, request: Request, group: Group) -> Response:
         :pparam Request request: the request object
         :pparam Group group: the group object
         """
+        serializer = RequestSerializer(data=request.query_params)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
         # The type of related issues to retrieve. Can be either `same_root_cause` or `trace_connected`.
         related_type = request.query_params["type"]
         extra_args = {
```

Comment:
the extra_args and related_type should then be pulled out of the serializer and passed along

you'll want to change the type of the dictionary received by the implementations -- probably to take named arguments instead of a dict

you'll also be able to clean up the `int(...)` call I introduced as well